src/routes/notifications/guards/notifications.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { AuthController } from '@/routes/auth/auth.controller';
+import { AuthGuard } from '@/routes/auth/guards/auth.guard';
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Request } from 'express';
+
+@Injectable()
+export class NotificationsGuard implements CanActivate {
```

Comment:
Small nit: what about renaming this to a more general term, like `OptionalAuthGuard`, `OptionalAccessTokenGuard` or simply `AccessTokenGuard`. I guess this guard could be used outside the "notifications" scope, wdyt?

Reply:
I renamed, moved and added an isolated test for it in da9bf557fe0afb1b1eabea3681f878250e7c90ee.